src/lib.rs

```diff
@@ -363,16 +366,15 @@ impl<const N: usize, T: Gene> System<N, T> {
 
     fn migrate_agents(&mut self) {
         let mut rng = thread_rng();
```

Comment:
This is not spacific to this commit, but this shouldn't use `thread_rng`. perhaps the system needs an rng of it's own for this situation.

Reply:
nevermind, fixed it in a separate PR